test/org/sagebionetworks/bridge/services/ScheduledActivityServiceMockTest.java

```diff
@@ -372,111 +380,154 @@ public void deleteActivitiesForSchedulePlanRejectsBadValue() {
     }
 
-    
     @Test
     public void newAndExistingActivitiesAreMergedV3() {
-        List<ScheduledActivity> scheduled = createNewActivities("AAA", "BBB", "CCC");
-        List<ScheduledActivity> db = createStartedActivities("AAA","CCC");
-
-        List<ScheduledActivity> saves = merge(scheduled, db, V3_MERGE);
-        assertActivities(saves, "BBB");
+        SchedulePlan aaa = schedulePlan("AAA");
+        SchedulePlan bbb = schedulePlan("BBB");
+        SchedulePlan ccc = schedulePlan("CCC");
         
-        scheduled = service.orderActivities(scheduled, V3_FILTER);
-        assertActivities(scheduled, "AAA", "BBB", "CCC");
+        when(schedulePlanService.getSchedulePlans(ClientInfo.UNKNOWN_CLIENT, TEST_STUDY)).thenReturn(Lists.newArrayList(aaa,bbb,ccc));
+        List<ScheduledActivity> db = Lists.newArrayList(createFinishedActivities("AAA"+TIME_PORTION).get(0),
+                createStartedActivities("BBB"+TIME_PORTION).get(0));
+        when(activityDao.getActivities(eq(TIME_ZONE), any())).thenReturn(db);
         
-        ScheduledActivity activity = scheduled.stream()
-                .filter(act -> act.getGuid().equals("AAA")).findFirst().get();
+        List<ScheduledActivity> returnedActivities = service.getScheduledActivities(createScheduleContext(NOW).build());
+        assertActivityGuids(returnedActivities, "BBB", "CCC");
+        
+        ScheduledActivity activity = returnedActivities.stream()
+                .filter(act -> { return act.getGuid().startsWith("BBB"); }).findFirst().get();
         assertTrue(activity.getStartedOn() > 0L);
+        
+        verify(activityDao).saveActivities(scheduledActivityListCaptor.capture());
+        List<ScheduledActivity> saves = scheduledActivityListCaptor.getValue();
+        assertActivityGuids(saves, "CCC");
     }
     
     @Test
     public void newAndExistingActivitiesAreMergedV4() {
-        List<ScheduledActivity> scheduled = createNewActivities("AAA", "BBB", "CCC");
-        List<ScheduledActivity> db = createStartedActivities("AAA","CCC");
-
-        List<ScheduledActivity> saves = merge(scheduled, db, V4_MERGE);
-        assertActivities(saves, "BBB");
+        SchedulePlan aaa = schedulePlan("AAA");
+        SchedulePlan bbb = schedulePlan("BBB");
+        SchedulePlan ccc = schedulePlan("CCC");
+        
+        when(schedulePlanService.getSchedulePlans(ClientInfo.UNKNOWN_CLIENT, TEST_STUDY)).thenReturn(Lists.newArrayList(aaa,bbb,ccc));
+        
+        List<ScheduledActivity> db = createStartedActivities("AAA"+TIME_PORTION,"CCC"+TIME_PORTION);
+        when(activityDao.getActivityHistoryV2(HEALTH_CODE, "BBB", NOW, NOW, TIME_ZONE, null, API_MAXIMUM_PAGE_SIZE))
+                .thenReturn(new ScheduledActivityList(db, null, 0));
+        
+        List<ScheduledActivity> returnedActivities = service.getScheduledActivitiesV4(createScheduleContext(NOW).build());
+        assertActivityGuids(returnedActivities, "AAA", "BBB", "CCC");
```

Comment:
This test is functionally identical to the existing test, but it doesn't seem to test that AAA and CCC were replaced by their started counterparts.